field/src/batch_inverse.rs

```diff
@@ -82,18 +90,25 @@ pub fn batch_multiplicative_inverse<F: Field>(x: &[F]) -> Vec<F> {
     for i in (WIDTH..n).rev() {
         // buf[i - WIDTH] has not been written to by this loop, so it equals
         // x[i % WIDTH] * x[i % WIDTH + WIDTH] * ... * x[i - WIDTH].
-        buf[i] = buf[i - WIDTH] * a_inv[i % WIDTH];
+        result[i] = result[i - WIDTH] * a_inv[i % WIDTH];
         // buf[i] now holds the inverse of x[i].
         a_inv[i % WIDTH] *= x[i];
     }
     for i in (0..WIDTH).rev() {
-        buf[i] = a_inv[i];
+        result[i] = a_inv[i];
     }
 
-    for (&bi, &xi) in buf.iter().zip(x) {
+    for (&bi, &xi) in result.iter().zip(x) {
         // Sanity check only.
         debug_assert_eq!(bi * xi, F::one());
     }
+}
 
-    buf
+fn batch_multiplicative_inverse_small<F: Field>(x: &[F], result: &mut [F]) {
+    // Currently we don't care much about speed on small inputs, so just do it naively.
+    x.par_iter()
```

Comment:
Maybe add an assert that x.len() < 4 or something (And if it's larger suggest using batch_multiplicative_inverse_helper).

Would be quite expensive if this accidentally gets run on a large list.